tools/cli.js

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env node
+
+'use strict';
+
+// TODO: support jstp://server and jstps://server
+
+const jstp = require('.');
+const readline = require('readline');
+
+const commandProcessor = {};
+const lineProcessor = {};
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+  completer
+});
+
+const log = msg => {
+  const userInput = rl.line;
+  if (userInput) rl.clearLine();
+  console.log(msg);
+  rl.write('\n');
+  if (userInput) rl.write(userInput);
+};
+
+rl.on('line', (line) => {
+  const [type, leftover] = _split(line.trim(), ' ', 1);
+  if (!type) {
+    return rl.prompt(true);
+  }
+  const processor = lineProcessor[type];
+  if (!processor) {
+    log(`Unknown command ${type}`);
+  } else {
```

Comment:
I'd prefer either the if branch to return from the function and reduce the indentation level for the code inside the else branch or to flip the condition. I don't feel very strongly for one way or other, but I like the first one more.